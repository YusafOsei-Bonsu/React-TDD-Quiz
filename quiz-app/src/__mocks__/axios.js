import mockAxios from 'jest-mock-axios';
export default {
    get: jest.fn(()=> Promise.resolve({data:{}}))
};

