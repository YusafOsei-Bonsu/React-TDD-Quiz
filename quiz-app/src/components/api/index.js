import axios from 'axios';
export const getCatData = (dispatch) => {
    return (dispatch) => {
        axios.get("https://opentdb.com/api_category.php").then(response => {
            const data = response.data;
            dispatch({ type: 'catDropDown', payload: data.trivia_categories });
        })

    }
};
// Fetch for quiz by category id, level and 
export const getQuizData = (value, dif, dispatch) => {
    return (dispatch) => {
        axios.get(`https://opentdb.com/api.php?amount=10&category=${value}&difficulty=${dif}&type=multiple`).then(async response => {
            const data = response.data.results;
            await dispatch({ type: 'questionsAnswers', payload: data });
        }).catch(() => dispatch({type: 'handle-error' }))
    }
};