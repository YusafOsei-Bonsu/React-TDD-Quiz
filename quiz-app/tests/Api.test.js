import mockAxios from 'jest-mock-axios';
import axios from 'axios';
import { getCatData, getQuizData } from '../src/components/api/index';

describe('Test API calling functions', ()=> {
    describe('Should be called', () => {
        it('getQuizData should accept arguments', () => {
            let value = "Art"
            let dif = "Easy"
          expect(getQuizData(value, dif)).toMatchSnapshot()
        })
        it('getCatData should accept dispatch', () => {
            let dispatch = jest.fn()
            
          expect(getCatData(dispatch)).toMatchSnapshot()
        })
    })
})