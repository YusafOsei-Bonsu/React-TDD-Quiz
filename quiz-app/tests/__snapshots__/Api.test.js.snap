// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tes Reducer should return the initial state 1`] = `[Function]`;

exports[`Tes Reducer should return the initial state 2`] = `[Function]`;

exports[`Test API calling  Reducer should return the initial state 1`] = `[Function]`;

exports[`Test API calling  Reducer should return the initial state 2`] = `[Function]`;

exports[`Test API calling functions S should return the initial state 1`] = `[Function]`;

exports[`Test API calling functions S should return the initial state 2`] = `[Function]`;

exports[`Test API calling functions Should be called getCatData should accept dispatch 1`] = `[Function]`;

exports[`Test API calling functions Should be called getCatData should return the initial state 1`] = `[Function]`;

exports[`Test API calling functions Should be called getQuiz 1`] = `[Function]`;

exports[`Test API calling functions Should be called getQuizData  1`] = `[Function]`;

exports[`Test API calling functions Should be called getQuizData should accept ar 1`] = `[Function]`;

exports[`Test API calling functions Should be called getQuizData should accept arguments 1`] = `[Function]`;

exports[`Test API calling functions Should be called should return the initial state 1`] = `[Function]`;

exports[`Test api Reducer should return the initial state 1`] = `[Function]`;

exports[`Test api Reducer should return the initial state 2`] = `[Function]`;

exports[`register Reducer should return the initial state 1`] = `[Function]`;

exports[`register Reducer should return the initial state 2`] = `[Function]`;
